miso-web/src/main/webapp/scripts/box_visualization.js

```diff
@@ -264,113 +260,94 @@ Box.DialogVisual = function(scan) {
-    });
-  };
+  
   return self;
 };
 
 Box.Visual = function() {
   var self = new BoxVisual();
 
   //@Override
-  self.getBoxItem = function(row, col) {
+  self.getBoxPositionOpts = function(row, col) {
     var pos = Box.utils.getPositionString(row, col);
     var boxables;
     boxables = self.data.filter(function(item) { return item.coordinates == pos; });
     if (boxables.length > 0) {
-      return new BoxItem({
-        row: row,
-        col: col,
-        selected: self.selected,
+      return {
         title: boxables[0].alias,
         selectedImg: '/styles/images/tube_full_selected.png',
-        unselectedImg: '/styles/images/tube_full.png',
-        onClick: function() {
-          jQuery('#selectedPosition').text(pos);
-          jQuery('#selectedName').text(boxables[0].name);
-          jQuery('#selectedAlias').html(Box.utils.hyperlinkifyBoxable(boxables[0].name, boxables[0].id, boxables[0].alias));
-          jQuery('#selectedName').html(Box.utils.hyperlinkifyBoxable(boxables[0].name, boxables[0].id, boxables[0].name));
-          jQuery('#selectedBarcode').val(boxables[0].identificationBarcode);
-          jQuery('#selectedBarcode').select().focus();
-          Box.ui.filterTableByColumn('#listingBoxablesTable', Box.utils.getPositionString(row, col), 0);
-          jQuery('#removeSelected, #emptySelected').prop('disabled', false).removeClass('disabled');
-          jQuery('#currentLocation, #currentLocationText, #warningMessages').html('');
-        }
-      });
+        unselectedImg: '/styles/images/tube_full.png'
+      };
     } else {
-      return new BoxItem({
-        row: row,
-        col: col,
-        selected: self.selected,
+      return {
         title: 'Empty',
         selectedImg: '/styles/images/tube_empty_selected.png',
-        unselectedImg: '/styles/images/tube_empty.png',
-        onClick: function() {
-          jQuery('#selectedPosition').text(Box.utils.getPositionString(row, col));
-          jQuery('#selectedName').empty();
-          jQuery('#selectedAlias').empty();
-          jQuery('#selectedBarcode').val('');
-          jQuery('#selectedBarcode').focus();
-          jQuery('#updateSelected, #removeSelected, #emptySelected').prop('disabled', true).addClass('disabled');
-          jQuery('#currentLocation, #currentLocationText, #warningMessages').html('');
-        }
-      });
+        unselectedImg: '/styles/images/tube_empty.png'
+      };
     }
   };
-
-  //@Override
-  self.getBoxPosition = function(row, col, tCell) {
-    return new BoxPosition({
-      row: row,
-      col: col,
-      cell: tCell,
-      boxItem: self.getBoxItem(row, col)
+  
+  self.onSelectionChanged = function(items) {
+    var positions = items.map(function(item) {
+      return Box.utils.getPositionString(item.row, item.col);
     });
+    Box.ui.filterTableByBoxPositions(positions);
+    
+    var boxables = null;
+    
+    if (positions.length === 1 && boxables && boxables.length > 0) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Change this condition so that it does not always evaluate to "false"; some subsequent code is never executed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2583)